### PR TITLE
Deprecate Windows x86-32 JEP Proposal

### PR DESCRIPTION
**Pull request by gdams**
This is still WIP and subject to changes

### CODE REVIEW

File: deprecate_32_bit/Deprecate_32-bit_Proposal.md [hunk 1]
```diff
@@ -0,0 +1,60 @@
+Summary
+-------
+ 
+Deprecate the Windows x86 32-bit ports, with the intent to remove them in a future release.
```
Comment (karianna) [inline deprecate_32_bit/Deprecate_32-bit_Proposal.md:4]:
```suggestion
Deprecate the Windows x86 32-bit port, with the intent to remove it in a future release.
```

**Comment by karianna** [comment]
I think ports -> port throughout

**Review by karianna — CHANGES_REQUESTED** [review]

File: deprecate_32_bit/Deprecate_32-bit_Proposal.md [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+1. Mark the ports, and related port-specific features, as deprecated for removal in the relevant JDK documentation.
+
+Non-Goals
+---------
+1. It is not a goal of this JEP to change the status of the affected ports in any prior release. The earliest release to which this JEP could be targeted is JDK 21.
+
+ 
+Motivation
+----------
+Dropping support for these ports will enable contributors in the OpenJDK Community to accelerate the development of new features that will move the platform forward. Windows 10 (the last Windows operating system to support a 32-bit installation) will reach EOL on October 14, 2025. With Java 21 being potentially supported up to September 2031, it makes sense to deprecate the 32-bit port from the codebase.
```
Comment (brunoborges) [inline deprecate_32_bit/Deprecate_32-bit_Proposal.md]: > With Java 21 being potentially supported up to September 2031, it makes sense to deprecate the 32-bit port from the codebase.

This line here is not required. Keep in mind that OpenJDK is the reference implementation of the Java specifications. It is not in anyway a commercially supported product, and the statement "Java 21 being potentially supported" without mentioning a specific commercial vendor, is a moot point for the purpose of this JEP.

Perhaps a better point of view is that we believe a major part of Java workloads on Windows 32-bit is running older versions of Java without any forecast for upgrading to modern, post OpenJDK 17, versions.

File: deprecate_32_bit/Deprecate_32-bit_Proposal_JEP.md [hunk 3]
```diff
@@ -0,0 +1,60 @@
+Summary
+-------
+ 
+Deprecate the Windows x86-32 port, with the intent to remove it in a future release.
+
+Goals
+-----
+The proposal has the following goals:
```
Comment (karianna) [inline deprecate_32_bit/Deprecate_32-bit_Proposal_JEP.md:8]:
```suggestion
The proposal has the following goals:

```

File: deprecate_32_bit/Deprecate_32-bit_Proposal_JEP.md [hunk 4]
```diff
@@ -0,0 +1,60 @@
+Summary
+-------
+ 
+Deprecate the Windows x86-32 port, with the intent to remove it in a future release.
+
+Goals
+-----
+The proposal has the following goals: 
+1. Enhance the build system to issue an error message when an attempt is made to configure a build for one of the deprecated port. The error message will be suppressible via a new configuration option.
```
Comment (karianna) [inline deprecate_32_bit/Deprecate_32-bit_Proposal_JEP.md:9]:
```suggestion
1. Enhance the build system to issue an error message when an attempt is made to configure a build for the deprecated port. The error message will be suppressible via a new configuration option.
```

File: deprecate_32_bit/Deprecate_32-bit_Proposal_JEP.md [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+1. Mark the port, and related port-specific features, as deprecated for removal in the relevant JDK documentation.
+
+Non-Goals
+---------
+1. It is not a goal of this JEP to change the status of the affected port in any prior release. The earliest release to which this JEP could be targeted is JDK 21.
+
```
Comment (karianna) [inline deprecate_32_bit/Deprecate_32-bit_Proposal_JEP.md:15]:
```suggestion
```

File: deprecate_32_bit/Deprecate_32-bit_Proposal_JEP.md [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+ 
+Motivation
+----------
+Dropping support for this port will enable contributors in the OpenJDK Community to accelerate the development of new features that will move the platform forward. Windows 10 (the last Windows operating system to support a 32-bit installation) will reach EOL on October 14, 2025, and I believe a major part of Java workloads on Windows 32-bit is running older versions of Java without any forecast for upgrading to modern, post OpenJDK 17, versions.
```
Comment (karianna) [inline deprecate_32_bit/Deprecate_32-bit_Proposal_JEP.md:19]:
```suggestion
Dropping support for this port will enable contributors in the OpenJDK Community to accelerate the development of new features that will move the platform forward. Windows 10 (the last Windows operating system to support a 32-bit installation) will reach EOL on October 14, 2025, and Microsoft believes that Java workloads on Windows 32-bit are running older versions of Java without any plans for upgrading to modern (OpenJDK 17+) versions.
```

File: deprecate_32_bit/Deprecate_32-bit_Proposal_JEP.md [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+Another motivation factor is that there is currently no implementation of [JEP 436 (Virtual Threads)](https://openjdk.org/jeps/436) for 32-bit platforms and without a vendor stepping forward to implement this it's unlikelty that vendors will be able to continue supporting the platform.
```
Comment (karianna) [inline deprecate_32_bit/Deprecate_32-bit_Proposal_JEP.md:20]:
```suggestion
Another motivation is that there is currently no implementation of [JEP 436 (Virtual Threads)](https://openjdk.org/jeps/436) for 32-bit platforms and without a vendor stepping forward to implement this it's unlikelty that vendors will be able to continue supporting the platform.
```

Comment (karianna) [inline deprecate_32_bit/Deprecate_32-bit_Proposal_JEP.md:20]:
```suggestion
Another motivation factor is that there is currently no implementation of [JEP 436 (Virtual Threads)](https://openjdk.org/jeps/436) for 32-bit platforms and without a vendor stepping forward to implement this it's unlikely that OpenJDK will be able to continue supporting the platform.
```

File: deprecate_32_bit/Deprecate_32-bit_Proposal_JEP.md [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+The main focus of this JEP is to deprecate the Windows x86 port but this JEP could also extend deprecate Linux at the same time.
```
Comment (karianna) [inline deprecate_32_bit/Deprecate_32-bit_Proposal_JEP.md:21]:
```suggestion
The main focus of this JEP is to deprecate the Windows x86 port.
```

**Review by karianna — CHANGES_REQUESTED** [review]

**Review by karianna — APPROVED** [review]

**Pull request merged**